File: dishpan/enamel.py
# enamel 

import json

datafile = open('../data/geoJSON/WhereWeWalk_March_2014.geojson', 'r')
data = json.loads(datafile.read())

def properties(timestamp):
	if "time" in timestamp["properties"]:
		return { "time": timestamp["properties"]["time"] }
	else:
		return { 
			"startTime": timestamp["properties"]["startTime"], 
			"endTime": timestamp["properties"]["endTime"]
		}

for feature in data["features"]:
	# timestampList = [{
	# 	"type": "Feature", 
	# 	"geometry": { 
	# 		"type": "Point", 
	# 		"coordinates": timestamp["geometry"]["coordinates"] 
	# 	}, 
	# 	"properties": properties(timestamp)
	# } for timestamp in feature["properties"]["timestamps"] ]
	# timestamps = {
	# 	"type": "FeatureCollection",
	# 	"features": timestampList
	# }
	timestampList = [timestamp["geometry"]["coordinates"][0] for timestamp in feature["properties"]["timestamps"]]
	timestamps = {
		"type": "MultiPoint",
		"coordinates": timestampList
	}



	newFeature = { 
		"type": "FeatureCollection", 
		"features": [
			feature,
			{
				"type": "FeatureCollection", 
				"features": timestamps
			}	
		]
	}
	pathFile = open('../data/geoJSON/package/path-' + str(feature["properties"]["database-id"]) + '.geojson', 'w+')
	timestampFile = open('../data/geoJSON/package/timestamp-' + str(feature["properties"]["database-id"]) + '.geojson', 'w+')
	json.dump(feature, pathFile);
	json.dump(timestamps, timestampFile)

File: dishpan/enamel.js
var Terraformer	= require('terraformer'),
	ArcGIS 		= require('terraformer-arcgis-parser'),
	fs 			= require('fs');

fs.readFile('/var/www/ozma/data/geoJSON/WhereWeWalk_March_2014.geojson', { encoding: 'utf8' }, function(err, data) {
	// console.log(JSON.parse(data));
	var features = new Terraformer.FeatureCollection(JSON.parse(data));
	var arcGISFeatures = ArcGIS.convert(features);
	arcGISFeatures.forEach(function(feature) { feature.attributes.OBJECTID = feature.attributes['database-id']; });
	console.log(JSON.stringify(arcGISFeatures));
	// console.log(arcGISFeatures);
});
